File: java/src/main/java/com/alxarafe/app/application/usecase/GetGreetingUseCase.java
package com.alxarafe.app.application.usecase;

import com.alxarafe.app.domain.model.Greeting;
import com.alxarafe.app.domain.model.GreetingRepository;

import java.util.List;
import java.util.Optional;

public final class GetGreetingUseCase {
    private final GreetingRepository repository;

    public GetGreetingUseCase(GreetingRepository repository) {
        this.repository = repository;
    }

    public Optional<Greeting> execute(String id) {
        return repository.findById(id);
    }

    public List<Greeting> findAll() {
        return repository.findAll();
    }
}

File: php/src/Application/UseCase/GetGreetingUseCase.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\UseCase;

use Alxarafe\App\Domain\Model\Greeting;
use Alxarafe\App\Domain\Model\GreetingRepositoryInterface;

final readonly class GetGreetingUseCase
{
    public function __construct(
        private GreetingRepositoryInterface $repository,
    ) {
    }

    public function execute(string $id): ?Greeting
    {
        return $this->repository->findById($id);
    }

    /**
     * @return Greeting[]
     */
    public function findAll(): array
    {
        return $this->repository->findAll();
    }
}
